sunpy/timeseries/metadata.py

```diff
@@ -327,10 +348,47 @@ def columns(self):
         for metatuple in self.metadata:
             all_cols.update(metatuple[1])
         return list(all_cols)
-        
-    def remove_columns(self, colnames):
+
+    @property
+    def metas(self):
+        """Returns a list of all the metadict objects in the TimeSeriesMetaData object."""
+        all_metas = []
+        for metatuple in self.metadata:
+            all_metas.append(metatuple[2])
+        return all_metas
+
+    @property
+    def timeranges(self):
+        """Returns a list of all the TimeRange objects the TimeSeriesMetaData object."""
+        all_tr = []
+        for metatuple in self.metadata:
+            all_tr.append(metatuple[0])
+        return all_tr
+
+    def values(self):
+        """Returns a list of all the values from the metadict objects in each
+        entry in the TimeSeriesMetaData object."""
+        all_vals = set()
+        for metatuple in self.metadata:
+            for key, value in metatuple[2].items():
+                all_vals.add(str(value))
+        all_vals = list(all_vals)
+        all_vals.sort()
+        return all_vals
+
+    @property
+    def timerange(self):
+        """Returns the TimeRange of the entire time series meta data."""
+        start = self.metadata[0][0].start

```

Comment:
The append method always adds in order. The only way I know of to make it not in order it to use the constructor with a list of tuple entries that are not in order. This is something I should check in the TSMD._validate_meta method and/or TS._sanitize_metadata method.